testutil/metrictest/assertions.go

```diff
@@ -0,0 +1,236 @@
+// Package metrictest is used for making assertions on metrics
+// produced by linkerd proxy.
+package metrictest
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/linkerd/linkerd2/testutil/metrictest/promm"
+)
+
+var (
+	addrRE      = regexp.MustCompile(`[0-9\.]+:[0-9]+`)
+	ipRE        = regexp.MustCompile(`[0-9\.]+`)
+	authorityRE = regexp.MustCompile(`[a-zA-Z\-]+\.[a-zA-Z\-]+\.svc\.cluster\.local:[0-9]+`)
+)
+
+// HasNoOutboundHTTPRequest returns error if there is any
+// series matching request_total{direction="outbound"}
+func HasNoOutboundHTTPRequest(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction": promm.Equals("outbound"),
+	})
+	ok, err := m.HasMatchInString(metrics)
+	if err != nil {
+		return fmt.Errorf("failed to run a check of against the provided metrics: %w", err)
+	}
+	if ok {
+		return fmt.Errorf("expected not to find HTTP outbound requests \n%s", metrics)
+	}
+	return nil
+}
+
+// HasOutboundHTTPRequestWithTLS checks there is a series matching:
+// request_total{
+//   direction="outbound",
+//   target_addr=~"[0-9\.]+:[0-9]+",
+//   target_ip=~"[0-9.]+",
+//   tls="true",
+//   dst_namespace="default.${ns}.serviceaccount.identity.linkerd.cluster.local",
+//   dst_serviceaccount="default"
+// }
+func HasOutboundHTTPRequestWithTLS(metrics, ns string) error {
+	m := promm.NewMatcher("request_total", promm.Labels{
+		"direction":          promm.Equals("outbound"),
+		"target_addr":        promm.Like(addrRE),
+		"target_ip":          promm.Like(ipRE),
+		"tls":                promm.Equals("true"),
+		"server_id":          promm.Equals(fmt.Sprintf("default.%s.serviceaccount.identity.linkerd.cluster.local", ns)),
```

Comment:
I think this is fine for now since we'll only use this in the opaqueports test. We should probably (in the long run) not assume the tls identity is `default`, but I think we can tweak this when there'll be a need for it.

Reply:
Yes, it will be interesting to see how this code can be made reusable between different tests. But for now, I was focused on showing how `promm` can (hopefully) make tests easier to understand.

Reply:
Yeah, agreed, let's keep it simpler for now.